src/mlpack/tests/hmm_test.cpp

```diff
@@ -904,66 +859,84 @@ BOOST_AUTO_TEST_CASE(GaussianHMMPredictTest)
  */
 BOOST_AUTO_TEST_CASE(GMMHMMPredictTest)
 {
-  // We will use two GMMs; one with two components and one with three.
-  std::vector<GMM> gmms(2);
-  gmms[0] = GMM(2, 2);
-  gmms[0].Weights() = arma::vec("0.75 0.25");
+  // It's possible, but extremely unlikely, that this test can fail.  So we are
+  // willing to do three trials in case the first two fail.
+  bool success = false;
+  for (size_t trial = 0; trial < 2; ++trial)
```

Comment:
"do three trails", but we only do 2 here, maybe I missed something? I guess since this is unlikely anyway, 2 are just fine.

Reply:
Oh, thanks.  Let me change that to `3` for extra paranoia.